src/main/java/scrolls/elder/logic/commands/LogAddCommand.java

```diff
@@ -76,14 +91,42 @@ public CommandResult execute(Model model) throws CommandException {
         Person befriendee = lastShownBList.get(befriendeeIndex.getZeroBased());
         Person volunteer = lastShownVList.get(volunteerIndex.getZeroBased());
 
+        if (!befriendee.isPairedWith(volunteer) && !volunteer.isPairedWith(befriendee)) {
+            throw new CommandException(MESSAGE_PERSONS_NOT_PAIRED);
+        }
+
+        Person updatedBefriendee = createPersonWithTimeServed(befriendee, duration);
+        Person updatedVolunteer = createPersonWithTimeServed(volunteer, duration);
+
         Log toAdd =
             new Log(model.getDatastore(), volunteer.getPersonId(), befriendee.getPersonId(), duration, startDate,
                 remarks);
 
         logStore.addLog(toAdd);
+
+        personStore.setPerson(befriendee, updatedBefriendee);
+        personStore.setPerson(volunteer, updatedVolunteer);
+        personStore.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL);
         return new CommandResult(MESSAGE_SUCCESS);
     }
 
+    private Person createPersonWithTimeServed(Person p, int duration) {
```

Comment:
Do you wanna move this to `PersonFactory` instead? Thematically I think its a bit more consistent, and it reduces the additional imports you have to do here. We should probably make a proper `PersonBuilder` at some point to reduce all these factory methods.

Reply:
Yes that was one of the implementations I considered but I eventually settled on keeping the method in LogAddCommand as I thought it would be consistent with createEditedPerson/createEditedPair in EditCommand

Reply:
Ogey then its fine bah, I'll see if its possible to make it less cumbersome to use ltr on.